system/inc/system_sleep.h

```diff
@@ -49,6 +49,7 @@ int system_sleep(Spark_Sleep_TypeDef mode, long seconds, uint32_t param, void* r
 int system_sleep_pin(uint16_t pin, uint16_t mode, long seconds, uint32_t param, void* reserved);
 int system_sleep_pins(const uint16_t* pins, size_t pins_count, const InterruptMode* modes, size_t modes_count, long seconds, uint32_t param, void* reserved);
 int system_sleep_ext(const hal_sleep_config_t* config, hal_wakeup_source_base_t** reason, void* reserved);
+int system_sleep_ext_impl(const hal_sleep_config_t* config, hal_wakeup_source_base_t** reason, void* reserved);
```

Comment:
This also can be kept out of the header as it's an internal implementation detail.